### PR TITLE
gapis/service/path: Add ResolveConfig

### PR DESCRIPTION
**Pull request by ben-clayton**
Pass this to `Get()`, `Set()`, `Follow()` and `Find()`.
Propagate the ResolveConfig thoughout the resolve package.

This is used to specify the replay device for resolving state data pulled back from the GPU.

### CODE REVIEW

File: gapis/resolve/framebuffer_attachment.go [hunk 1]
```diff
@@ -39,7 +39,9 @@ func FramebufferAttachment(
 	attachment api.FramebufferAttachment,
 	settings *service.RenderSettings,
 	hints *service.UsageHints,
+	config *path.ResolveConfig,
```
Comment (pmuetschard) [inline gapis/resolve/framebuffer_attachment.go]: This a bit redundant now. Both the `ReplaySettings` and the `ResolveConfig` contain the replay device.

Should we also remove the default device selection below?

Comment (ben-clayton) [reply gapis/resolve/framebuffer_attachment.go]: I want to move this RPC into a Get. At that point, yes, we should certainly de-dupe this field.
I haven't done this yet as I wanted to make sure these changes worked first.

Comment (pmuetschard) [reply gapis/resolve/framebuffer_attachment.go]: SGTM

**Comment by ben-clayton** [comment]
Merged in https://github.com/google/gapid/pull/2130

**Pull request closed without merging**